src/main/java/com/talentpath/airTNB/services/ListingService.java

```diff
@@ -32,6 +32,17 @@ public Listing addListing(Listing toAdd) {
         return listingRepository.saveAndFlush(toAdd);
     }
 
+    public List<Listing> getListingsByCityAndState(Integer listingId) throws NullListingException {
+        Optional<Listing> listing = listingRepository.findById(listingId);
+        String city, state;
```

Comment:
String city and state are never used, can be assigned and refactored to

String city = listing.get().getCity();
String state = listing.get().getState();
listingRepository.findAllByCityIgnoreCaseAndStateIgnoreCase(city,state);